cmd/govim/main.go

```diff
@@ -197,6 +197,14 @@ type govimplugin struct {
 	applyEditsCh   chan applyEditCall
 	applyEditsLock sync.Mutex
 
+	// popupMenus is a map of active popups where the user can make a selection. It is keyed
+	// by popupID and the value is a function that will be called when the popup is closed.
+	// It is either closed by selecting an item (1-indexed selection is passed as argument),
+	// by calling popup_close() (0 is passed as argument) or ESC is pressed (-1 is passed as
+	// argument. Adding and removing popups is protected by popupMenusLock.
+	popupMenus     map[int]func(selection int) error
+	popupMenusLock sync.Mutex
```

Comment:
Can you clarify why a lock is required here? Because it appears that all work with the map happens on the vimstate "thread"